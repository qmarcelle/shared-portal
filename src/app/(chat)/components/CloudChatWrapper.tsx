'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
