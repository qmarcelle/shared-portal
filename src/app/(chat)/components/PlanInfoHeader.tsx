'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
