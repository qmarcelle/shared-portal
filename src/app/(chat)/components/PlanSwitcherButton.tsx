'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
