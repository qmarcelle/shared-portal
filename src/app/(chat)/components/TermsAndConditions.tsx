'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
