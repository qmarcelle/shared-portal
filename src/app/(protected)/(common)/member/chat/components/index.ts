/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
