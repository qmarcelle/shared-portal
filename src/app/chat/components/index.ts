// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
