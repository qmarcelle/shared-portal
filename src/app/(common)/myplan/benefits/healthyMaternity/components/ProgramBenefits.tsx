/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
