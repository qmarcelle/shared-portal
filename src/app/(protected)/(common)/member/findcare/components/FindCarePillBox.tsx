import { IComponent } from '@/components/IComponent';
import { Button } from '@/components/foundation/Button';
import { Card } from '@/components/foundation/Card';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';

import { Header } from '@/components/foundation/Header';

interface PillBoxProps extends IComponent {
  className: string;
  title: string;
  icon?: JSX.Element;
  pillObjects: FindCarePillBoxData[];
}

export interface FindCarePillBoxData {
  label: string;
  callback?: () => void | Promise<void> | null;
}

export const FindCarePillBox = ({
  className,
  title,
  pillObjects,
  icon,
}: PillBoxProps) => {
  return (
    <Card className={className}>
      <section className="gap-8">
        <Row className="align-top items-center">
          {icon}
          <Spacer axis="horizontal" size={16} />
          <Header text={title} type="title-3" className="!font-bold" />
        </Row>
        <Spacer size={16} />
        <Row className="flex-wrap gap-x-2 gap-y-4">
          {pillObjects.map((pillData, index) => (
            <Button
              key={index}
              type="pill"
              label={pillData.label}
              callback={pillData.callback}
              className="h-[28px] font-bold primary-color"
            />
          ))}
        </Row>
      </section>
    </Card>
  );
};
