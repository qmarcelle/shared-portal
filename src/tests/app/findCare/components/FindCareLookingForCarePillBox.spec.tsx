import { FindCarePillBox } from '@/app/(protected)/(common)/member/findcare/components/FindCarePillBox';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Image from 'next/image';
import findCareIcon from '/assets/find_care_search.svg';

const renderUI = () => {
  return render(
    <FindCarePillBox
      className="md:w-[480px] md:h-[200px] md:my-8 p-4 w-11/12 sm:m-4"
      title="Looking for care? Find a:"
      icon={<Image src={findCareIcon} className="w-[40px] h-[40px]" alt="" />}
      pillObjects={[
        {
          label: 'Primary Care Provider',
          callback: () => {
            console.log('Clicked Pill PCP');
          },
        },
        {
          label: 'Dentist',
          callback: () => {
            console.log('Clicked Pill Dentist');
          },
        },
        {
          label: 'Mental Health Provider',
          callback: () => {
            console.log('Clicked Pill Mental Health Provider');
          },
        },
        {
          label: 'Eye Doctor',
          callback: () => {
            console.log('Clicked Pill Eye Doctor');
          },
        },
        {
          label: 'Pharmacy',
          callback: () => {
            console.log('Clicked Pill Pharmacy');
          },
        },
        {
          label: 'Virtual Care',
          callback: () => {
            console.log('Clicked Pill Virtual Care');
          },
        },
      ]}
    />,
  );
};

describe('FindCarePillBox', () => {
  it('should render UI correctly', () => {
    const component = renderUI();

    screen.getByRole('heading', {
      name: 'Looking for care? Find a:',
    });

    screen.getByLabelText('Primary Care Provider');
    screen.getByLabelText('Dentist');
    screen.getByLabelText('Mental Health Provider');
    screen.getByLabelText('Eye Doctor');
    screen.getByLabelText('Pharmacy');
    screen.getByLabelText('Virtual Care');
    expect(component.baseElement).toMatchSnapshot();
  });
});
