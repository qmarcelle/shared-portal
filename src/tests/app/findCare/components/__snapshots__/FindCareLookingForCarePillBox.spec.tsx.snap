// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindCarePillBox should not call sso if Pharamcybenifits visibility rule false on click of Pharmacy 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col h-175px w-full text-white justify-center items-center brand-gradient px-4 "
      >
        <div
          class="flex flex-col items-start w-full app-content px-4"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <h1
            class="title-1 font-thin"
          >
             
            Find Care & Costs
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-1 undefined"
          >
            <span>
              Looking for care or estimating costs for a procedure? Use our
               
            </span>
            <span
              class="link !text-white"
            >
              <a>
                Find Care & Costs tool.
              </a>
            </span>
          </span>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-2 undefined"
          >
            <span>
              In case of medical emergency, call 911. In case of a mental health crisis, call 988.
            </span>
          </span>
        </div>
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
      />
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main mt-4"
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          />
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;

exports[`FindCarePillBox should not call sso if visionBenefits visibility rule true on click of vision 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col h-175px w-full text-white justify-center items-center brand-gradient px-4 "
      >
        <div
          class="flex flex-col items-start w-full app-content px-4"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <h1
            class="title-1 font-thin"
          >
             
            Find Care & Costs
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-1 undefined"
          >
            <span>
              Looking for care or estimating costs for a procedure? Use our
               
            </span>
            <span
              class="link !text-white"
            >
              <a>
                Find Care & Costs tool.
              </a>
            </span>
          </span>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-2 undefined"
          >
            <span>
              In case of medical emergency, call 911. In case of a mental health crisis, call 988.
            </span>
          </span>
        </div>
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
      />
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main mt-4"
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          />
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;

exports[`FindCarePillBox should render UI correctly 1`] = `
<body>
  <div>
    <main
      class="flex flex-col justify-center items-center page"
    >
      <div
        class="flex flex-col h-175px w-full text-white justify-center items-center brand-gradient px-4 "
      >
        <div
          class="flex flex-col items-start w-full app-content px-4"
        >
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
          <h1
            class="title-1 font-thin"
          >
             
            Find Care & Costs
          </h1>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-1 undefined"
          >
            <span>
              Looking for care or estimating costs for a procedure? Use our
               
            </span>
            <span
              class="link !text-white"
            >
              <a>
                Find Care & Costs tool.
              </a>
            </span>
          </span>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
          />
          <span
            class="rich-text body-2 undefined"
          >
            <span>
              In case of medical emergency, call 911. In case of a mental health crisis, call 988.
            </span>
          </span>
        </div>
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
      />
      <div
        class="flex flex-col app-content app-base-font-color md:p-0 p-4"
      >
        <section
          class="card-main mt-4"
        >
          <div
            class="flex flex-row items-center input   "
          >
            <input
              class="body-1 search-textfield"
              name="search-Field"
              placeholder="Search by procedure, specialty, facility or provider name..."
              type="text"
              value=""
            />
            <div
              class="flex flex-col items-end"
            >
              <button
                aria-label="Search"
                class="flex flex-col p-[0.125rem]  "
                type="submit"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  class="icon searchhover "
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </button>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Looking for care? Find a:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Primary Care Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Primary Care Provider
                  </button>
                  <button
                    aria-label="Dentist"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dentist
                  </button>
                  <button
                    aria-label="Mental Health Provider"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Mental Health Provider
                  </button>
                  <button
                    aria-label="Eye Doctor"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Eye Doctor
                  </button>
                  <button
                    aria-label="Pharmacy"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Pharmacy
                  </button>
                  <button
                    aria-label="Virtual Care"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Virtual Care
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="flex flex-col flex-grow page-section-36_67 items-stretch"
          >
            <div
              class="card-main my-8  p-4"
            >
              <section
                class="gap-8"
              >
                <div
                  class="flex flex-row align-top items-center"
                >
                  <img
                    alt=""
                    class="w-[40px] h-[40px]"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                  <span
                    style="display: block; width: 16px; min-width: 16px; height: 1px; min-height: 1px;"
                  />
                  <h3
                    class="title-3 !font-bold"
                  >
                    Planning for a procedure? Estimate costs for:
                  </h3>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                />
                <div
                  class="flex flex-row flex-wrap gap-x-2 gap-y-4"
                >
                  <button
                    aria-label="Medical"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Medical
                  </button>
                  <button
                    aria-label="Dental"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Dental
                  </button>
                  <button
                    aria-label="Prescription Drugs"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Prescription Drugs
                  </button>
                  <button
                    aria-label="Vision"
                    class="button-text pill text-center flex flex-row justify-center items-center min-w-fit null h-[28px] font-bold primary-color"
                    type="button"
                  >
                    Vision
                  </button>
                </div>
              </section>
            </div>
          </div>
        </section>
        <section
          class="flex flex-row items-start app-body mt-4"
        >
          <div
            class="flex flex-col flex-grow page-section-63_33 items-stretch"
          />
        </section>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
      </div>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
      />
    </main>
  </div>
</body>
`;
